### DISCUSSION TITLE
Persistent Plugins while using Docker

### DISCUSSION THREAD

**Discussion by cyberndj** [netbox-community/netbox-docker#788] [Getting Help] [answered]
Howdy,

My organization uses Docker to run Netbox. We installed some plugins that we like (and work with the current version). However, we noticed than anytime the container gets re-created (like after a reboot) we have to run the pip install commands in the venv and restart the container to get Netbox and plugins to work. 

I have the plugins in the local-requirements file. I've also tried creating a custom docker-file with RUN commands to execute the PIP commands, but I get a wheel error and it bombs.

Do you have any ideas on how to make the plugins persistent or how to add some PIP commands to the startup before Netbox starts?

Thanks!
-Jake

**Comment by kkthxbye-code** [comment, marked as answer]
Did you try following the documentation?

https://github.com/netbox-community/netbox-docker/wiki/Using-Netbox-Plugins

**Reply by cyberndj** [reply]
Howdy,
Adding to the complexity, we use Portainer to manage our docker server. I'll have to look at the portainer docs too to see where to save the files so everyone in our org can update as necessary. 

-I also looked at running the install command using the exec function in the compose/(portainer stack) file, but it errored with file/folder not found. So I think docker runs the exec function before mounting the voumes.

**Comment by cyberndj** [comment]
Hi,
There are limitations with Portainer and docker compose where you can't specifiy a local folder/file to copy into the image. --So that is a bust.

The quickest and most reliable way I've seen is to follow the docs.
